File: backend/internal/domain/business/businessService.go
package business

// Service provides business-related operations.
type Service struct {
	repository Repository
}

// NewService creates a new instance of the Business Service with the given repository.
func NewService(repo Repository) *Service {
	return &Service{repository: repo}
}

// CreateBusiness handles the logic to create a new business.
func (s *Service) CreateBusiness(name, owner, contactInfo string) (*Business, error) {
	business := &Business{Name: name, Owner: owner, ContactInfo: contactInfo}
	err := s.repository.Save(business)
	if err != nil {
		return nil, err
	}
	return business, nil
}

// GetBusiness retrieves a business by ID.
func (s *Service) GetBusiness(id int) (*Business, error) {
	return s.repository.FindByID(id)
}

// GetAllBusinesses retrieves all businesses.
func (s *Service) GetAllBusinesses() ([]*Business, error) {
	return s.repository.FindAll()
}

// DeleteBusiness deletes a business by ID.
func (s *Service) DeleteBusiness(id int) error {
	return s.repository.Delete(id)
}

// UpdateBusiness updates the information of an existing business.
func (s *Service) UpdateBusiness(id int, name, owner, contactInfo string) (*Business, error) {
	// Update the business directly using the Update method
	updatedBusiness, err := s.repository.Update(id, name, owner, contactInfo)
	if err != nil {
		return nil, err
	}

	return updatedBusiness, nil
}

File: frontend/src/domain/business/services/businessService.js
// src/domain/business/services/businessService.js

export const fetchBusinessData = async (businessId, setFetchedData) => {
    if (!businessId) {
      alert("Please enter a valid business ID.");
      return;
    }
    try {
      const response = await fetch(`${process.env.REACT_APP_API_URL}/business/${businessId}`);
      if (!response.ok) throw new Error("Failed to fetch data");
      const data = await response.json();
      setFetchedData(data);
    } catch (error) {
      console.error(error);
    }
  };


  export const getAllBusinesses = async (setBusinesses, setError) => {
    try {
      const response = await fetch(`${process.env.REACT_APP_API_URL}/businessAll`);
      if (!response.ok) {
        // Parse and log the error message
        const errorData = await response.json();
        console.error("Error fetching businesses:", errorData.error);
        setError(errorData.error); // Optionally set an error state to display in UI
        return;
      }
  
      const data = await response.json();
      setBusinesses(data);
    } catch (error) {
      console.error("Network error fetching businesses:", error);
      setError("Network error: Unable to retrieve businesses");
    }
  };
  


  export const createBusiness = async (business, setBusinessData, onComplete) => {
    try {
      const response = await fetch(`${process.env.REACT_APP_API_URL}/business`, {
        method: "POST",
        body: JSON.stringify(business),
        headers: { "Content-Type": "application/json" },
      });
      if (!response.ok) throw new Error("Failed to create business");
  
      const data = await response.json();
      setBusinessData(data); // Update with created business data
      alert("Business created successfully");
  
      if (onComplete) onComplete(); // Trigger the callback after successful creation
    } catch (error) {
      console.error(error);
    }
  };
  

  export const updateBusiness = async (businessId, business, setBusinessData) => {
    if (!businessId || !business.name || !business.owner || !business.contact_info) {
      alert("Please fill out all fields.");
      return;
    }
  
    try {
      const response = await fetch(`${process.env.REACT_APP_API_URL}/business/${businessId}`, {
        method: "PUT",
        body: JSON.stringify(business),
        headers: { "Content-Type": "application/json" },
      });
  
      if (!response.ok) throw new Error("Failed to update business");
      const data = await response.json();
      setBusinessData(data); // Updates parent component's state
      alert("Business updated successfully");
    } catch (error) {
      console.error(error);
    }
  };
  

export const deleteBusiness = async (businessId, setBusinessData) => {
  if (!businessId) {
    alert("Please enter a valid business ID to delete.");
    return;
  }

  try {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/business/${businessId}`, {
      method: "DELETE",
    });

    if (!response.ok) throw new Error("Failed to delete business");
    alert("Business deleted successfully");
    setBusinessData(null); // Clear business data after deletion
  } catch (error) {
    console.error(error);
  }
};

  
  